e2e/cypress/integration/markdown/markdown_image_spec.js

```diff
@@ -46,13 +47,33 @@ describe('Markdown', () => {
         // * Verify that HTML Content is correct.
         // Note we use the Gigantic timeout to ensure that the large images will load
         cy.getLastPostId().then((postId) => {
-            cy.get(`#postMessageText_${postId}`).find('p').
-                should('have.text', 'GitHub favicon:  ');
+            cy.get(`#postMessageText_${postId}`).find('div').
+                should('have.text', 'GitHub favicon:  ').
+                and('have.class', 'style--none');
 
             cy.get(`#postMessageText_${postId}`).find('img').
                 should('have.class', 'markdown-inline-img').
+                and('have.class', 'markdown-inline-img--hover').
+                and('have.class', 'cursor--pointer').
+                and('have.class', 'a11y--active').
                 and('have.attr', 'alt', 'Github').
                 and('have.attr', 'src', `${baseUrl}/api/v4/image?url=https%3A%2F%2Fgithub.githubassets.com%2Ffavicon.ico`);
         });
     });
+
+    it('opens preview window when image is clicked', () => {
+        // #  Post markdown message
+        cy.postMessageFromFile('markdown/markdown_inline_images_2.md');
+
+        // Note we use the Gigantic timeout to ensure that the large images will load
+        cy.getLastPostId().then((postId) => {
+            cy.get(`#postMessageText_${postId}`, {timeout: TIMEOUTS.GIGANTIC}).then((post) => {
+                // # Get the image and simulate a click.
+                cy.wrap(post).find('img.markdown-inline-img').first().click();
```

Comment:
Please add height/width check just to make sure that the image is loaded before clicking.  (Note that we have case where this img element still renders even if there's an error occurred.)

Reply:
Also, you mentioned the 3 cases:
1. If the image source is unsafe, a link is rendered
2. If the image source is safe, the image will have a pointer cursor & a hover effect, and clicking on it will open the preview window
3. If the image is a link itself, the image will be rendered and clicking on it will open the link in a new browser window (no preview or hover effect)

Not blocking but would be nice if we have E2E for those.

Reply:
Thanks @saturninoabril 

Yep, case `1.` is not covered. Thought unit testing is enough for it, but will add it here as well.

`2.` the test named `opens preview window when image is clicked` covers the interactivity aspect of it, and test `with in-line images 2` mostly covers its static aspect/existence. Maybe it is best to combine them together in one test covering the whole use case?

`3.` the test named `with in-line images 1` covers it except for clicking the link and navigating to its destination, which i think should not be the focus of this test suit. However, happy to do so if you would like to test link clicking as well, let me know. Should be renamed as well, right?

Initially, i tried not to do massive changes to the existing tests, I guess that didn't help much 🙃

Reply:
1. If unit test covered, then it's all good.
2. Oh okay, then it's good as is (no need to combine)
3. Yup, I agree with it.  Good as is.

Thanks for looking into it!